### ISSUE TITLE
Rust analyzer yells when including an `Option` type in a custom enum type 

### ISSUE THREAD

**Issue by kalepail** [stellar/rs-soroban-sdk#1044]
### What version are you using?
`0.9.1`

### What did you do?

```rust
#[contracttype]
#[derive(Clone, Debug, Eq, PartialEq)]
pub enum Test {
    Poof(Option<Address>, Address, i128)
}
```

### What did you expect to see?
A happy clappy code editor

### What did you see instead?
![Screenshot 2023-07-28 at 1 26 44 PM](https://github.com/stellar/rs-soroban-sdk/assets/4383610/8f9d6036-1ae7-4568-9dc7-dc20fdde8063)
```
the trait bound `core::option::Option<soroban_sdk::Address>: SorobanArbitrary` is not satisfied
the following other types implement trait `SorobanArbitrary`:
  ()
  ContractContext
  CreateContractHostFnContext
  Glyph
  GlyphType
  HashType
  InvokerContractAuthEntry
  OfferType
and 23 others
```

**Comment by tyvdh** [comment]
Note I can still run `soroban contract build` so it's not an actual error

**Comment by dmkozh** [comment]
Doesn't seem like this happens anymore (which is expected given that 1 year has passed).

**Comment by kalepail** [comment]
I'm still seeing this issue with `21.4.0`
![CleanShot 2024-07-31 at 09 12 48@2x](https://github.com/user-attachments/assets/4071a949-e69d-408c-a7f7-876782ec0dc4)

**Comment by dmkozh** [comment]
This issue is pretty hard to reproduce consistently and it *seems* like it doesn't occur with p22 SDK, at least I haven't been able to trigger it (at the same time I was able to trigger it with p21 SDK again). I won't close this yet, but will double-check after v22 SDK release.

**Comment by leighmcculloch** [comment]
It looks like this was the fix. When I cherry pick that fix onto 21.7.4, `cargo test`, works just fine.
- https://github.com/stellar/rs-soroban-sdk/pull/1338

I've opened a backport release 21.7.5 that contains the fix, as the fix is easily cherry-picked onto the sdk.
- https://github.com/stellar/rs-soroban-sdk/pull/1370

@dmkozh Could you review the release PR, and I'll do a release once approved.

**Comment by leighmcculloch** [comment]
Fixed in release v21.7.5:
- https://github.com/stellar/rs-soroban-sdk/pull/1370
